src/cli.cr

```diff
@@ -2,7 +2,80 @@ require "dppm/cli"
 require "./dppm_rest_api"
 
 module DppmRestApi::CLI
-  def self.run_server(config, host, port, data_dir, **args)
+  extend self
+
+  # A helper method for the select_users method -- splits a comma-separated
+  # list of numbers in a string into an `Array` of `Int32` values.
+  private def split_groups_arg(argument text : String) : Array(Int32)
+    text.split(',').map do |g|
+      if g_int = g.to_i?
+        g_int
+      else
+        abort "\
+          group ID #{g} wasn't a number, please use numeric group ids \
+          separated by commas (,)."
+      end
+    end
+  end
+
+  # All variables are received by a method as a `String | Nil`. This macro
+  # converts an argument, by name, to its `String` value, or raises an
+  # error instructing the user to include the variable value.
+  private macro required(arg)
+    mn_{{arg.id}} || abort "{{ arg.id.gsub /_/, "-" }} argument is required."
+  end
+
+  # select the users from the current configuration according to the name,
+  # groups the user is a member of, and/or the user's API key.
+  private def select_users(match_name, match_groups, api_key, from selected_users)
+    # Filter by name
+    if name = match_name
+      if name.starts_with?('/') && name.ends_with?('/')
```

Comment:
```suggestion
      if stripped_name = name.lchop?('/').try &.rchop?('/')
```

Reply:
*Very* nice. I had a feeling I didn't have the best solution.